### PR TITLE
Create report directory automatically

### LINKED ISSUE #43: Feature request: Automatic create report_dir
Currently, we need to create a directory specified in report_dir if it doesn't exists. So my proposal is try creating that directory before generating reports. 

Thoughts?

### PR DESCRIPTION
**Pull request by wingyplus**
Add a new configuration to create `:report_dir` directory when it's set
to true.

Closes #43

### CODE REVIEW

**Comment by victorolinasc** [comment]
That was fast :) THanks a lot!

Can you add this option to the README?

**Comment by wingyplus** [comment]
> That was fast :) THanks a lot!
> 
> Can you add this option to the README?

I add it. But I'm not good at writing English. Please kindly review. :)

File: test/formatter_test.exs [hunk 1]
```diff
@@ -317,6 +317,18 @@ defmodule FormatterTest do
       assert JUnitFormatter.get_report_file_path() ==
                "#{Mix.Project.app_path()}/junit_formatter-report_file_test.xml"
     end
+
+    test "create directory at init" do
```
Comment (victorolinasc) [inline test/formatter_test.exs:321]: Can we just create another test that will run init with the dir already created?

**Comment by victorolinasc** [comment]
Sorry it took so long :) I think we are fine with this... just one test nitpick

**Comment by wingyplus** [comment]
@victorolinasc Sorry for the late response. I fix the code from your suggestion and add more tests including check `File.mkdir_p!` should match with `:ok` to make sure it can create. Please kindly review.

**Review by victorolinasc — APPROVED** [review]

**Comment by victorolinasc** [comment]
Thanks and sorry for the delay! I will be releasing it now.

**Comment by wingyplus** [comment]
@victorolinasc thank you 😊

**Pull request merged**